internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixCustomizedViewOfflinePushRepository.java

```diff
@@ -40,9 +43,52 @@ public class HelixCustomizedViewOfflinePushRepository extends HelixBaseRoutingRe
   private final ReentrantReadWriteLock resourceAssignmentRWLock = new ReentrantReadWriteLock();
   private static final String LEADER_FOLLOWER_VENICE_STATE_FILLER = "N/A";
 
-  public HelixCustomizedViewOfflinePushRepository(SafeHelixManager manager) {
+  private final Map<String, Integer> resourceToPartitionCountMap = new VeniceConcurrentHashMap<>();
+
+  private final ReadOnlyStoreRepository storeRepository;
+
+  public HelixCustomizedViewOfflinePushRepository(SafeHelixManager manager, ReadOnlyStoreRepository storeRepository) {
     super(manager);
     dataSource.put(PropertyType.CUSTOMIZEDVIEW, Collections.singletonList(HelixPartitionState.OFFLINE_PUSH.name()));
+    this.storeRepository = storeRepository;
+    this.storeRepository.registerStoreDataChangedListener(new StoreDataChangedListener() {
+      @Override
+      public void handleStoreCreated(Store store) {
+        int currentVersion = store.getCurrentVersion();
+        if (currentVersion == NON_EXISTING_VERSION) {
+          return;
+        }
+
+        String newResourceName = Version.composeKafkaTopic(store.getName(), currentVersion);
+        int partitionCount = store.getVersion(currentVersion).get().getPartitionCount();
+        resourceToPartitionCountMap.put(newResourceName, partitionCount);
+      }
+
+      @Override
+      public void handleStoreChanged(Store store) {
+        int currentVersion = store.getCurrentVersion();
+        if (currentVersion == NON_EXISTING_VERSION) {
+          return;
+        }
+
+        resourceToPartitionCountMap.entrySet().removeIf(entry -> {
+          String storeName = Version.parseStoreFromKafkaTopicName(entry.getKey());
+          int version = Version.parseVersionFromVersionTopicName(entry.getKey());
+          return store.getName().equals(storeName) && version != currentVersion;
```

Comment:
Is it correct to keep only the current version? Do we not care for the other existing (but non-current) versions? I am not sure...

Reply:
For CV, does future version matter? backup most definitely not.  To be safe I will just delete non-existent versions.